lib/lightning/scrubber.ex

```diff
@@ -88,15 +96,18 @@ defmodule Lightning.Scrubber do
   Prepare a list of sensitive samples (strings) into a potentially bigger list
   composed of variations a sample may appear.
   """
-  @spec encode_samples(samples :: [String.t()]) :: [String.t()]
-  def encode_samples(samples) do
+  @spec encode_samples(samples :: [String.t()], usernames :: [String.t()]) :: [
+          String.t()
+        ]
+  def encode_samples(samples, usernames) do
     stringified_samples =
       samples
       |> Enum.filter(fn x -> not is_boolean(x) end)
       |> Enum.map(fn x -> if is_integer(x), do: Integer.to_string(x), else: x end)
 
     base64_secrets =
       stringified_samples
+      |> Enum.concat(usernames)
       |> cartesian_pairs()
       |> Enum.map(fn [x, y] ->
         "#{x}:#{y}"
```

Comment:
Username or email needed for matching Base64 auth method (Basic authentication) as tested here:
https://github.com/OpenFn/platform-app/blob/4d4b1d35d8abc491757ee192d8c22b26ebb80573/test/services/credential_service_test.exs#L73